analyzers/tests/SonarAnalyzer.UnitTest/Rules/Hotspots/DisablingRequestValidationTest.cs

```diff
@@ -39,33 +39,41 @@ public class DisablingRequestValidationTest
     {
         private const string AspNetMvcVersion = "5.2.7";
         private const string WebConfig = "Web.config";
+        private const string SonarProjectConfig = "SonarProjectConfig.xml";
         // for the tests that don't test the XML logic, to avoid test failures caused by unexpected issues from Web.config files
-        private const string FolderWithoutWebConfig = @"TestCases\WebConfig\Empty\";
+        private const string ProjectConfigTemplate = @"
+<SonarProjectConfig xmlns=""http://www.sonarsource.com/msbuild/analyzer/2021/1"">
+    <FilesToAnalyzePath>{0}\FilesToAnalyze.txt</FilesToAnalyzePath>
+</SonarProjectConfig>";
 
         [TestMethod]
         [TestCategory("Rule")]
         public void DisablingRequestValidation_CS() =>
             Verifier.VerifyAnalyzer(@"TestCases\DisablingRequestValidation.cs",
-                new CS.DisablingRequestValidation(AnalyzerConfiguration.AlwaysEnabled, FolderWithoutWebConfig),
+                new CS.DisablingRequestValidation(AnalyzerConfiguration.AlwaysEnabled),
                 additionalReferences: NuGetMetadataReference.MicrosoftAspNetMvc(AspNetMvcVersion));
 
         [TestMethod]
         [TestCategory("Rule")]
         public void DisablingRequestValidation_CS_Disabled() =>
             Verifier.VerifyNoIssueReported(@"TestCases\DisablingRequestValidation.cs",
-                new CS.DisablingRequestValidation(AnalyzerConfiguration.Hotspot, @"TestCases\WebConfig"),
+                new CS.DisablingRequestValidation(AnalyzerConfiguration.Hotspot),
                 additionalReferences: NuGetMetadataReference.MicrosoftAspNetMvc(AspNetMvcVersion));
 
         [DataTestMethod]
         [DataRow(@"TestCases\WebConfig\S5753Values")]
         [DataRow(@"TestCases\WebConfig\Foo")]
         [DataRow(@"TestCases\WebConfig\Corrupt")]
         [TestCategory("Rule")]
-        public void DisablingRequestValidation_CS_WebConfig(string root) =>
+        public void DisablingRequestValidation_CS_WebConfig(string root)
+        {
+            WriteSonarProjectConfigToDisk(root);
             DiagnosticVerifier.VerifyExternalFile(
                 SolutionBuilder.Create().AddProject(AnalyzerLanguage.CSharp).GetCompilation(),
-                new CS.DisablingRequestValidation(AnalyzerConfiguration.AlwaysEnabled, root),
-                File.ReadAllText(Path.Combine(root, WebConfig)));
+                new CS.DisablingRequestValidation(AnalyzerConfiguration.AlwaysEnabled),
+                File.ReadAllText(Path.Combine(root, WebConfig)),
+                Path.Combine(root, SonarProjectConfig));
```

Comment:
`WriteSonarProjectConfigToDisk(root);` does `Path.Combine(filesToAnalyzeDirectory, SonarProjectConfig)` inside and then is always followed by `Path.Combine(root, SonarProjectConfig)`. That's redundant.

Let's orchestrate the test like this and let the `Create` method return the full path.
```
DiagnosticVerifier.VerifyExternalFile(
                SolutionBuilder.Create().AddProject(AnalyzerLanguage.CSharp).GetCompilation(),
                new CS.DisablingRequestValidation(AnalyzerConfiguration.AlwaysEnabled),
                File.ReadAllText(Path.Combine(root, WebConfig)),
                CreateSonarProjectConfig(root));
```